### ISSUE TITLE
Wow! Finally! Thanks! 

### ISSUE THREAD

**Issue by jonerikandersen** [kayjei/adax_wifi#6]
Will give the implementation a go, and give you some feedback along the way :-)

**Comment by jonerikandersen** [comment]
Getting those heat signatures... Wow. I'll give it a massive go to get it from one heater. But I have four in total.

**Comment by kayjei** [comment]
Thanks!
Basically, you get the signatures per zone, not per radiator. Do you have them all in different zones?

**Comment by jonerikandersen** [comment]
Yes. Looks like that. It looks like it is not possible to have more than one heater in one zone?

**Comment by kayjei** [comment]
No, but you need to retreive signature for all temperature you want to be able to set, plus temperature 0 (which is used to turn it off.
My recommendation is to retrieve siganture for all temperatures between min and max ( set in app).
Please also note that you must use the dev-branch to be able to set multiple zones

**Comment by jonerikandersen** [comment]
Allready on it. At the moment, I get read temps from all through the zone list I guess. However, setting the temperatures, I need to add all four zones with all 30 temp signatures. 
Even though I think I only need 0+14 to 23. Using the dev-branch atm. But this looks great! I'm trying to bridge it into Homekit atm. : )

**Comment by joggs** [comment]
Have you succeeded in sniffing the adax app version 2? I am curious what the appVersion parameter value is and if the api response gives you different and hopefully more data.

**Comment by jchasey** [comment]
Grabbed the signatures just fine for heat and zone. 
For the temperature however, there does not appear to be an encoded signature, just the temperature encoded into the url? Could you give an example of the form required?
Thanks!

**Comment by jchasey** [comment]
> Grabbed the signatures just fine for heat and zone.
> For the temperature however, there does not appear to be an encoded signature, just the temperature encoded into the url? Could you give an example of the form required?
> Thanks!

Sorted, was checking the wrong request, signatures found ok.

**Comment by kayjei** [comment]
> Have you succeeded in sniffing the adax app version 2? I am curious what the appVersion parameter value is and if the api response gives you different and hopefully more data.

@joggs my appversion is iOS-adax-2.4.0. I would say Yes, this should be version 2

**Comment by cortado** [comment]
You have an example of how the parameters.py should look. Have got all the values, but still no dice. I am wondering if I have some values with wrong syntax or notation.

**Comment by kayjei** [comment]
Sorry for my late reply.
Basically, the file looks like (dev-branch):
```python
def get_static(value):
    switcher = {
    "account_id": '12345',
    "appVersion": 'iOS-adax-2.4.0',
    "device": 'iPhone10,4',
    "os": 'iOS 2012.4',
    "timeOffset": '120',
    "timeZone": 'Europe/Stockholm',
    "locale": '',
    "zone_signature": '302C02142396D5FGFGAFG856468FDGCB4018FDF93F0214079A870064DBFE22691A27FC09F3AFE23B7A7B22'
    }
    return switcher.get(value,"Key missing")

def get_signature_456789(value):
    switcher = {
    "heat_signature": '302D0214035B337E5CDDC9EE3C59AA54FA688610215009740C147DAAB144859E592B9BC4493DBC2CAA004',
    0: '302D02146DB0C8AA23548666A666A69A81485434226B02150080BD1B0DE19DEABD3D8EE007AA09B93EB3BBBFBA', #Set as HVAC_OFF
    5: '302D02142572F806E126F9F4DEF0510AF7A9526ED3AC91D70215008AB85235A584FB928423C50E366261D4DD315B',
    }
    return switcher.get(value,"Key missing")
```
The signature key is normally 32 or 34 characters.
